File: practica2_839756_842236/c++/instruccionDup.cc
// Autores: Andrei Vlasceanu [839756] & Andres Yubero [842236]

#include "instruccionDup.h"

void InstruccionDup::ejecutar(stack<int> *s)
{
    int x;
    x = s->top();
    s->push(x);
}

void InstruccionDup::listar()
{
    cout << "dup" << endl;
}

File: practica2_839756_842236/java/InstruccionDup.java
import java.util.Stack;

public class InstruccionDup extends Instruccion {
    @Override
    public void ejecutar(Stack<Integer> s) {
        int x = s.peek();
        s.push(x);
    }

    @Override
    public void listar() {
        System.out.println("dup");
    }
}
